File: src/map.php
<?php

$nilMap = [];
echo count($nilMap) . PHP_EOL;
// echo $nilMap['abc']; // PHP Warning:  Undefined array key "abc"
$nilMap['abc'] = 'def'; // 書き込み可能

$teams = [
    'giants' => ['坂本', '岡本'],
    'tigers' => ['糸井', '大山'],
    'carp'   => ['鈴木', '安部'],
];
echo $teams['giants'][0] . PHP_EOL; // 坂本

$user = [
    'name' => 'Taro',
    'age' => 25,
];
echo $user['name'] . PHP_EOL; // Taro
unset($user['age']);
var_export($user);
echo PHP_EOL;

// 連想配列のキーの存在チェック
echo isset($user['name']) ? 'true' : 'false'; // true
echo PHP_EOL;
echo isset($user['age']) ? 'true' : 'false'; // false
echo PHP_EOL;

File: src/map.go
package main

import "fmt"

func main() {
	var nilMap map[string]int  // 初期値nil
	fmt.Println(len(nilMap))   // 0
	fmt.Println(nilMap["abc"]) // 0(intのゼロ値)
	//nilMap["abc"] = 3        // panic: nilマップへの書き込みはできない

	wins := map[string]int{} // string型のキーとint型の値を持つ空のマップ
	fmt.Println(wins == nil) // false
	fmt.Println(wins["abc"]) // 0(intのゼロ値)
	wins["abc"] = 3          // 書き込み可能
	fmt.Println(wins["abc"]) // 3

	teams := map[string][]string{
		"giants": {"坂本", "岡本"},
		"tigers": {"糸井", "大山"},
		"carp":   {"鈴木", "菊池"},
	}
	fmt.Println(teams["giants"][0]) // 坂本

	ages := make(map[string]int, 10) // サイズを指定してmapを作成
	fmt.Println(len(ages))           // 0

	user := map[string]any{
		"name": "Taro",
		"age":  25,
	}
	fmt.Println(user["name"]) // Taro
	delete(user, "age")
	fmt.Println(user)

	// マップのキーの存在チェック
	m := map[string]int{
		"apple":  100,
		"banana": 200,
	}

	v, ok := m["apple"]
	fmt.Println(v, ok) // 100 true

	v, ok = m["banana"]
	fmt.Println(v, ok) // 200 true

	v, ok = m["nothing"]
	fmt.Println(v, ok) // 0 false
}
